syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -367,10 +368,12 @@ def send(
 
             location = location[0]
 
-            if hasattr(self, "child") and isinstance(self.child, PointerTensor):
-                self.child.garbage_collect_data = False
-                if self._is_parameter():
-                    self.data.child.garbage_collect_data = False
+            if hasattr(self, "child"):
+
```

Comment:
why was this check split up? Alternatively, I think it might make sense to move both of these checks to a seperate function since it seems we write the combination alot